### PR TITLE
Fix(pager): Resolve background color before staring pager

### LINKED ISSUE #636: Unexpected behaviour using issue view on Ubuntu, Zsh
**Description**

When I execute `glab issue view 4` the result doesn't allow me to use up/down neither  j/k keys. When I exit from that view by pressing q+Enter, I have to press ctrl-c as well to efectivily return to the prompt. This only happens with a specific issue.

**Expected Behavior vs Actual Behavior**

I assume that `less` (or another similar program) is executed by piping the content to them. So pressing j/k is correct the way to scrolling there.

**Possible Fix**

**Steps to Reproduce**

1. Type this `glab issue 4` (my issue)
2. View the output 
3. See error

<details>
<summary>Here is the content of my issue</summary>

## Description

```bash
$ dag foo
foo
├── /home
├── /home
├── /home
├── /home
├── /home
└── /home


$ dag foo
foo


$ dag foo
foo
├── /home/
├── /home/
├── /home/
├── /home/
└── /home/


$ dag foo
foo
├── /home/
├── /home/
├── /home/
├── /home/
├── /home/
└── /home/


$ dag foo
foo
├── /home/
└── /home/


$ dag foo
foo


$ dag foo
foo


$ dag foo
foo


$ dag foo
foo
├── /home/
├── /home/
├── /home/
├── /home/
└── /home/


$ dag foo
foo


$ dag foo
foo
├── /home/
└── /home/


$ dag foo
foo
└── /home/

```

## Traceback error

Paste here the complete traceback error, including the command executed.

## Versions

- foo
- foo (only for installation problems)
- foo (only for installation problems)
- foo (only for installation problems)

```bash
# Paste the folling and copy the complete output here
pip freeze | grep foo
python -V
pip -V
```

### Others

(Optionally include the versions of other software that may concern)

</details>

**Logs**

**Your Environment**

* Version used (Run `glab --version`): `glab version 1.15.0 (2021-02-15)`
* Operating System and version: `Linux  5.4.0-66-generic #74-Ubuntu SMP Wed Jan 27 22:54:38 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`

### PR DESCRIPTION
**Pull request by zemzale**
**Description**
So far we have started the pager and then got background color glamour
style from enviorment.

If enviorment value is not set it defaults to auto. Which in turn uses
termenv to resolve background color using some terminal escape code
sequences. The problem with this is that doing this when a pager like
less is running, just doesn't work and leaves less in a broken state.

To fix this we are now checking the background color manually and before
we start the pager.

**Related Issue**

Resolves #636

**How Has This Been Tested?**
Fixes hangs for less on my machine.

**Types of changes**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation
- [ ] Chore (Related to CI or Packaging to platforms)

### CODE REVIEW

**Comment by profclems** [comment]
@zemzale kindly rebase the branch

**Comment by zemzale** [comment]
@profclems Will do. There seem to be some issues about this with tests anyways. So gotta fix that first

**Comment by zemzale** [comment]
Rebased and tests working

File: pkg/iostreams/iostreams.go [hunk 1]
```diff
@@ -166,6 +169,31 @@ func (s *IOStreams) IsOutputTTY() bool {
 	return s.IsErrTTY && s.IsaTTY
 }
 
+func (s *IOStreams) ResolveBackgroundColor() string {
+	style := os.Getenv("GLAMOUR_STYLE")
```
Comment (profclems) [inline pkg/iostreams/iostreams.go]:
```suggestion
func (s *IOStreams) ResolveBackgroundColor(style string) string {
        if(style == "") {
		style = os.Getenv("GLAMOUR_STYLE")
        }
```

File: commands/issue/view/issue_view.go [hunk 2]
```diff
@@ -93,6 +93,7 @@ func NewCmdView(f *cmdutils.Factory) *cobra.Command {
 				}
 			}
 
+			f.IO.ResolveBackgroundColor()
```
Comment (profclems) [inline commands/issue/view/issue_view.go]:
```suggestion
			f.IO.ResolveBackgroundColor(opts.GlamourStyle)
```

File: commands/mr/view/mr_view.go [hunk 3]
```diff
@@ -77,6 +77,7 @@ func NewCmdView(f *cmdutils.Factory) *cobra.Command {
 				}
 			}
 
+			f.IO.ResolveBackgroundColor()
```
Comment (profclems) [inline commands/mr/view/mr_view.go]:
```suggestion
			f.IO.ResolveBackgroundColor(glamourStyle)
```

**Review by profclems — CHANGES_REQUESTED** [review]
Thanks for working on this.

I have added a few suggestions in the inline review comments: we allow users set the `glamour_style` in the config so we should consider that. 
So in the `mr_view` and `issue_view`, we get the `glamour_style` from the config and pass to the `ResolveBackgroundColor(style string)`.

**Comment by zemzale** [comment]
Fixed the issue with not config-defined styles. 

I also removed the `GlamourStyle` variables being passed around since we really don't need that anymore, because we are resolving the style before, and it's available from IOStreams.

**Review by profclems — APPROVED** [review]
LGTM

**Pull request merged**